### PR TITLE
fix: [generic + sort] All ready for generics

### PR DESCRIPTION
**Pull request by tjgurwara99**
Beginning of Generics in Go. A relatively simple PR to include some examples use cases of Generics in sort. Was not able to actually convert some of the sort algorithms due to their intrinsic dependence on type `int`.

### CODE REVIEW

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Ok

**Review by raklaptudirm — DISMISSED** [review]
LGTM

File: sort/mergesort.go [hunk 1]
```diff
@@ -33,22 +36,22 @@ func merge(a []int, b []int) []int {
 
 }
 
-//Mergesort Perform mergesort on a slice of ints
-func Mergesort(items []int) []int {
+//Merge Perform mergesort on a slice
```
Comment (cclauss) [inline sort/mergesort.go:39]:
```suggestion
// Merge Perform mergesort on a slice
```

File: sort/quicksort.go [hunk 2]
```diff
@@ -33,8 +35,8 @@ func QuickSortRange(arr []int, low, high int) {
 	}
 }
 
-// QuickSort Sorts the entire array
-func QuickSort(arr []int) []int {
+// Quick Sorts the entire array
```
Comment (cclauss) [inline sort/quicksort.go:38]: quicksort is a single word without a space.

File: sort/sorts_test.go [hunk 3]
```diff
@@ -176,62 +177,62 @@ func benchmarkFramework(b *testing.B, f func(arr []int) []int) {
...
 }
 
 func BenchmarkCount(b *testing.B) {
-	benchmarkFramework(b, Count)
+	benchmarkFramework(b, sort.Count[int])
 }
 
 func BenchmarkQuick(b *testing.B) {
-	benchmarkFramework(b, QuickSort)
+	benchmarkFramework(b, sort.Quick[int])
```
Comment (cclauss) [inline sort/sorts_test.go:208]:
```suggestion
	benchmarkFramework(b, sort.Quicksort[int])
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Nice!

**Review by cclauss — APPROVED** [review]

**Review by siriak — APPROVED** [review]

File: sort/countingsort.go [hunk 4]
```diff
@@ -6,11 +6,13 @@
 
 package sort
 
-func Count(data []int) []int {
+import "github.com/TheAlgorithms/Go/constraints"
+
+func Count[T constraints.Number](data []int) []int {
```
Comment (quaeler) [inline sort/countingsort.go:11]: why isn't this
```
func Count[T constraints.Number](data []T) []T {
```
?

Comment (tjgurwara99) [reply sort/countingsort.go:11]: Primarily because I missed it... Feel free to open a PR to rectify it.

**Pull request merged**